src/08_example/05_tests.md

```diff
@@ -0,0 +1,53 @@
+# Testing the TCP Server
+Let's move on to testing our `handle_connection` function.
+First, we need a `TcpStream` to work with, but we don't want to make a real TCP connection in test code.
```

Comment:
Hmm, why not make a real connection? Seems like it could be a good integration/e2e test, checking that concurrent connections are possible for instance. No need to add such a test, but to discourage the strategy seems a bit strong.

Perhaps state that here we will write a unit test for the connection handler that will verify the contents of the response.